src/plugins/orangehrmRecruitmentPlugin/Controller/WorkflowActionHistoryController.php

```diff
@@ -50,4 +60,26 @@ public function preRender(Request $request): void
         $component->addProp(new Prop('history-id', Prop::TYPE_NUMBER, $historyId));
         $this->setComponent($component);
     }
+
+    public function isCapable(Request $request): bool
+    {
+        if ($request->attributes->getInt('candidateId') && $request->attributes->getInt('historyId')) {
+            $candidateId = $request->attributes->getInt('candidateId');
+            $historyId = $request->attributes->getInt('historyId');
+            if (!$this->getUserRoleManager()->isEntityAccessible(Candidate::class, $candidateId)) {
+                return false;
+            }
+            if (!$this->getUserRoleManager()->isEntityAccessible(CandidateHistory::class, $historyId)) {
+                return false;
+            }
+            if (!$this->getCandidateService()
+                    ->getCandidateDao()
+                    ->getCandidateHistoryRecordByCandidateIdAndHistoryId($candidateId, $historyId)
+                instanceof CandidateHistory) {
+                return false;
```

Comment:
I feel here it should be no record found

Reply:
when manually changing the url with history id, the 4x behaviour is it shows required credentials message, if the particular candidate does not have a record with the historyId. if we throw record not found, it keeps loading.